client/src/store/isLogin.js

```diff
@@ -0,0 +1,16 @@
+/* eslint-disable no-return-assign */
+/* eslint-disable no-param-reassign */
+
+import { createAction, createReducer } from '@reduxjs/toolkit';
+
+const initialState = {
+  isLogin: !!localStorage.getItem('token'),
+};
+
+export const toggleIsLogin = createAction('TOGGLE_IS_LOGIN');
+
+export default createReducer(initialState, {
+  [toggleIsLogin]: (state) => {
+    state.isLogin = !state.isLogin;
+  },
+});
```

Comment:
```suggestion
import { createSlice } from '@reduxjs/toolkit';

const initialState = {
  isLogin: !!localStorage.getItem('token'),
};

const userSlice = createSlice({
  name: 'user',
  initialState,

  reducers: {
    toggleIsLogin: (state) => {
      state.isLogin = !state.isLogin; //we mutate the state directly, immer works under the hood
    },
  },
});
// toggleIsLogin is an action creator created automatically by toolkit
export const { toggleIsLogin } = userSlice.actions;
export default userSlice.reducer;
```



this is the correct way to utilize Slices in redux toolkit

also the filename has to be changes to userSlice.js  and placed inside slices folder which has to be created inside the store folder

Reply:
done